### PR TITLE
Fetch Interrogated task reports

### PR DESCRIPTION
**Pull request by ajrgrubbs**
Added an endpoint to retrieve reports from
Interrogated. Since the reports are generally
huge, this includes filesystem caching.

This commit also servers as the first commit,
and therefore contains standard node.js files.

TO TEST:
You need a valid report ID to hit. I used 393. Others can be found in interrogated as necessary. Make sure to hit the same report twice to make sure there's no issues in the filesys caching.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const express = require('express')
```
Comment (adamyala) [inline index.js]: There are semi-colons missing in here and on a couple other lines. Can we get those added in?

**Comment by adamyala** [comment]
Looks good to me. Once you get those semi-colons added in, 👍

File: reports/reports.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+const fs = require('fs');
+const request = require('request-promise');
+const config = require('../config');
+
+var initialized = false;
+
+function getReport(reportId, response) {
+  init();
+  var filePath = `${REPORTS_DIR}/task_${reportId}.json`;
+  fs.stat(filePath, function(err, stat) {
```
Comment (adamyala) [inline reports/reports.js]: I don't think stat is ever referenced or used. Should we still keep it?

Comment (csavage1994) [inline reports/reports.js]: Agreed, unless you will be using it we might as well get rid of it, otherwise 👍

File: index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const express = require('express');
+const app = express();
+const port = 8080;
+
+const reports = require('./reports/reports');
+
+app.get('/', function (request, response) {
```
Comment (vforgione) [inline index.js]: Seems kind of superfluous to have this endpoint. Maybe set it up as a redirect to the latest report?

**Comment by vforgione** [comment]
👍

File: index.js (hunk 3 above)
Comment (ajrgrubbs) [inline index.js]: True, though this was just meant as a temporary placeholder. In my work for the next PR it's already been replaced by the MVP's main interface.

**Pull request merged**